samcli/commands/init/init_templates.py

```diff
@@ -0,0 +1,128 @@
+        return self._init_options_from_manifest(runtime, dependency_manager)
+
+    def _init_options_from_manifest(self, runtime, dependency_manager):
+        manifest_path = os.path.join(self.repo_path, "manifest.json")
+        with open(str(manifest_path)) as fp:
+            body = fp.read()
+            manifest_body = json.loads(body)
+            templates = manifest_body.get(runtime)
+            if templates is None:
+                # Fallback to bundled templates
+                return self._init_options_from_bundle(runtime, dependency_manager)
+            if dependency_manager is not None:
+                templates_by_dep = itertools.takewhile(lambda x: x["dependencyManager"] == dependency_manager, templates)
+                return list(templates_by_dep)
+            return templates
+
+    def _init_options_from_bundle(self, runtime, dependency_manager):
+        for mapping in list(itertools.chain(*(RUNTIME_DEP_TEMPLATE_MAPPING.values()))):
+            if runtime in mapping["runtimes"] or any([r.startswith(runtime) for r in mapping["runtimes"]]):
+                if not dependency_manager or dependency_manager == mapping["dependency_manager"]:
+                    mapping["appTemplate"] = "hello-world"  # when bundled, use this default template name
+                    return [mapping]
+        msg = "Lambda Runtime {} and dependency manager {} does not have an available initialization template.".format(
+            runtime, dependency_manager
+        )
+        raise UserException(msg)
+
+    def _clone_repo(self):
+        shared_dir = global_cfg.config_dir
+        expected_path = os.path.normpath(os.path.join(shared_dir, self._repo_name))
+        if self._should_clone_repo(expected_path):
+            try:
+                subprocess.check_output(["git", "clone", self._repo_url], cwd=shared_dir, stderr=subprocess.STDOUT)
+                self.repo_path = expected_path
+            except subprocess.CalledProcessError as clone_error:
+                output = clone_error.output.decode("utf-8")
+                if "not found" in output.lower():
+                    click.echo("WARN: Could not clone app template repo.")
+        self.clone_attempted = True
+
+    def _should_clone_repo(self, expected_path):
+        path = Path(expected_path)
+        if path.exists():
+            if not self._no_interactive:
+                overwrite = click.confirm("Init templates exist on disk. Do you wish to update?")
+                if overwrite:
+                    shutil.rmtree(expected_path)  # fail hard if there is an issue
+                    return True
+            self.repo_path = expected_path
+            return False
+        else:
+            if self._no_interactive:
+                return self._auto_clone
+            do_clone = click.confirm(
+                "This process will clone app templates from https://github.com/awslabs/aws-sam-cli-app-templates - is this ok?"
```

Comment:
do we prompt even in a non interactive mode here?

Reply:
We do not.

Reply:
ah I misread the indentation.